subsys/net/l2/wifi/wifi_shell.c

```diff
@@ -1671,14 +1671,17 @@ static int cmd_wifi_btm_query(const struct shell *sh, size_t argc, char *argv[])
 {
 	struct net_if *iface = net_if_get_first_wifi();
 	uint8_t query_reason = 0;
+	long tmp = 0;
 
 	context.sh = sh;
 
-	if (!parse_number(sh, (long *)&query_reason, argv[1], NULL,
+	if (!parse_number(sh, (long *)&tmp, argv[1], NULL,
```

Comment:
Since `tmp` is of type `long`, we no longer need the cast?

Reply:
Indeed, good point, will remove it.